hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHBaseComparators.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.storage;
+
+import org.apache.hadoop.hbase.KeyValue;
+
+/**
+ * This class is explicitly used as Key Comparator to work around the hard coded
+ * legacy format class names inside HBase. Otherwise, we will face issues with shading.
+ */
+public class HoodieHBaseComparators {
+  public static class HoodieHBaseKVComparator
```

Comment:
why nest this? just a simple top level class?

Reply:
Thought of keeping it static just like `HoodieKVComparator`.

Reply:
made it a top level class.